framework/src/base/MooseApp.C

```diff
@@ -47,6 +48,32 @@
 
 #define QUOTE(macro) stringifyName(macro)
 
+// Note: we need this two-part macro approach so that we can stringize
+// a macro *after* it's been expanded.
+// http://stackoverflow.com/questions/5256313/c-c-macro-string-concatenation
+// We are also relying on the fact that when configure defines
+// something, it explicitly defines it to be 1.
+
+// Turn A into a string literal without expanding macro definitions
+// (however, if invoked from a macro, macro arguments are expanded).
+#define STRINGIZE_NX(A) #A
+
+// Turn A into a string literal after macro-expanding it.
+#define STRINGIZE(A) STRINGIZE_NX(A)
```

Comment:
This one may be new. Maybe we just need to put these in `MooseTypes.h` and quite replicating them around.